packages/astro/src/vite-plugin-astro-server/response.ts

```diff
@@ -57,6 +57,12 @@ export async function writeWebResponse(res: http.ServerResponse, webResponse: Re
 
 	const _headers = Object.fromEntries(headers.entries());
 
+	// Undici 5.20.0+ includes a `getSetCookie` helper that returns an array of all the `set-cookies` headers.
+	// Previously, `headers.entries()` would already have these merged, but it seems like this isn't the case anymore.
+	if ('getSetCookie' in headers && typeof headers.getSetCookie === 'function') {
```

Comment:
Not important since this is a dev server, but this is an annoying thing about TS where it wants you to use `in` which is slow, when just a directly `typeof headers.getSetCookie === 'function'` is what you really want to do here. That's why I tend to pair `typeof` with `as any`, ala `typeof (something as any).prop`. Just needed to rant on that, I'll walk myself out now.

Reply:
Fair! That's a great thing to keep in mind, but yes I was doing it for the type safety